stake-pool/js/rollup.config.js

```diff
@@ -31,95 +31,85 @@ function generateConfig(configType, format) {
-      '@solana/spl-token',
-      '@solana/web3.js',
-      'bn.js',
-      'buffer',
-    ];
-  }
+  if (browser) {
+    if (format === 'iife') {
+      config.external = ['http', 'https'];
 
-  switch (configType) {
-    case 'browser':
-      switch (format) {
-        case 'esm': {
-          config.output = [
-            {
-              file: 'dist/index.browser.esm.js',
-              format: 'es',
-              sourcemap: true,
-            },
-          ];
-
-          // Prevent dependencies from being bundled
-          config.external = [
-            '@project-serum/borsh',
-            '@solana/buffer-layout',
-            '@solana/spl-token',
-            '@solana/web3.js',
-            'bn.js',
-            'buffer',
-          ];
-
-          break;
-        }
-        case 'iife': {
-          config.external = ['http', 'https'];
-
-          config.output = [
-            {
-              file: 'dist/index.iife.js',
-              format: 'iife',
-              name: 'solanaStakePool',
-              sourcemap: true,
-            },
-            {
-              file: 'dist/index.iife.min.js',
-              format: 'iife',
-              name: 'solanaStakePool',
-              sourcemap: true,
-              plugins: [terser({ mangle: false, compress: false })],
-            },
-          ];
-
-          break;
-        }
-        default:
-          throw new Error(`Unknown format: ${format}`);
-      }
-
-      // TODO: Find a workaround to avoid resolving the following JSON file:
-      // `node_modules/secp256k1/node_modules/elliptic/package.json`
-      config.plugins.push(json());
-
-      break;
-    case 'node':
       config.output = [
         {
-          file: 'dist/index.cjs.js',
+          file: 'dist/index.iife.js',
+          format: 'iife',
+          name: 'solanaStakePool',
+          sourcemap: true,
+        },
+        {
+          file: 'dist/index.iife.min.js',
+          format: 'iife',
+          name: 'solanaStakePool',
+          sourcemap: true,
+          plugins: [terser({ mangle: false, compress: false })],
+        },
+      ];
+    }
+    else {
+      config.output = [
+        {
+          file: 'dist/index.browser.cjs.js',
           format: 'cjs',
           sourcemap: true,
         },
         {
-          file: 'dist/index.esm.js',
+          file: 'dist/index.browser.esm.js',
           format: 'es',
           sourcemap: true,
         },
       ];
-      break;
-    default:
-      throw new Error(`Unknown configType: ${configType}`);
+
+      // Prevent dependencies from being bundled
+      config.external = [
+        '@project-serum/borsh',
+        '@solana/buffer-layout',
+        '@solana/spl-token',
+        '@solana/web3.js',
+        'bn.js',
+        'buffer',
+      ];
+    }
+
+    // TODO: Find a workaround to avoid resolving the following JSON file:
+    // `node_modules/secp256k1/node_modules/elliptic/package.json`
+    config.plugins.push(json());
```

Comment:
ESM and CJS build _should_ be able to use identical config for everything except the `format`.